docs/adr/adr-0018-how-to-give-feedback-after-a-user-action.md

```diff
@@ -0,0 +1,414 @@
+
+We also had doubts about where to place the role attribute:
+
+* place it on the element dynamically added to the DOM (the `<cc-toast>` element)
+* or place it on the container of the dynamically added content (the `<cc-toaster>` element).
+
+#### Test and support
+
+To take a decision, we needed to conduct fairly extensive tests to check how screen readers support the different combinations.
+You can find the detail of these tests in the section bellow.
+
+##### Dynamically inserted live regions
+
+Our tests show that `role` attributes seem to work best when they are not inserted dynamically.
+In other words, it seems more robust to use the `role` attribute on the `<cc-toaster>` element instead of the `<cc-toast>` element.
+The former is always present in the DOM while the latter is inserted dynamically.
+
+##### Role status
+
+Unfortunately, we found that the support for the `role="status"` was very inconsistent and insufficient (Nothing is announced with JAWS + Firefox)
+
+It is also interesting to note that with some combinations of screen reader + browser (NVDA + Firefox and Talkback + Chrome), it was announced as an `aria-atomic="false"`, meaning only new content inserted in the area was announced.
+
+##### Role log
+
+The `role="log"` attribute suffers a big issue with VoiceOver (MacOS) + Safari.
+The screen reader only announces the differences between content that is already present in the live region and content that has been inserted. For instance:
+
+* the live region contains a toast "Hello World number 1",
+* we insert an additional message "Hello Word number 2",
+* It announces "2".
+
+The excepted behavior would have been to announce "Hello World number 2".
+
+We faced the same issue when using `aria-live="polite"` with `aria-atomic="false"`.
+
+#### Decision
+
+Considering all this, we decided to use the most robust method to simulate a `role="status"` live region by using both `aria-live="polite` and `aria-atomic="true"`.
+
+We may reconsider this when we add a way for users to review the last toast messages, to make it less verbose but this will mostly depend on the support of the `role="log"` live regions by then, especially with VoiceOver and Safari (MacOS).
+
+## Going further
+
+What did we learn?
+
+As we mentioned before, we had to make some compromises on the accessibility.
+We understood that we could not do anything magical that would make the toast pattern fully accessible.
+We are now convinced that the toast pattern has its limitation and should not be used everytime.
+For example, it doesn't suit for the following cases:
+
+* when there is an important action to be added along.
+* when the message is very important.
+* when the message is too long.
+
+### Actions inside a toast
+
+We decided to avoid any actions inside toast as it comes with too much accessibility issues.
+Some people may not have the time to reach it while some other may not see it at all.
```

Comment:
Just to show that this sentence is related to the previous one: "For instance, some people may not have enough time..."